modules/tasks/taskClaim.js

```diff
@@ -22,38 +25,43 @@ const sendConfirmationEmail = (task, user, comments) => {
   )
 }
 
-const verifyIssueAndClaim = async (task, user, comments) => {
+const verifyIssueAndClaim = async (task, user, comments, token) => {
   const language = user.language || 'en'
   i18n.setLocale(language)
 
-  if (task.provider === 'github' && user.provider === task.provider) {
-    if (task.metadata.user === user.provider_username) {
-      const taskUser = task.user
-
-      // Update task
-      await taskUpdate({ id: task.id, userId: user.id })
-
-      const body = `${i18n.__('mail.task.claim.confirmation.body', {
-        title: task.title,
-        url: task.url,
-        comments: comments
-      })}`
+  return jwt.verify(token, process.env.SECRET_PHRASE, async (err, decoded) => {
+    // the 401 code is for unauthorized status
+    if (err || decoded !== task.id) {
```

Comment:
```suggestion
    if (err || parseInt(decoded) !== parseInt(task.id)) {
```
We need to check both variables converting to numbers, in my case, it was comparing a string and a number